### PR TITLE
CI test for the RNN

### PR DESCRIPTION
**Pull request by jsschreck**
Added the ci test for the rnn model
Fixed a bug in the rnn training script
Removed an erroneous "torch" directory in geckoml that created a problem with the ci test

### CODE REVIEW

File: .github/workflows/train_and_run.yml [hunk 1]
```diff
@@ -28,11 +28,19 @@ jobs:
...
+      - name: Run the gecko train script for the MLP model
         run: |
           conda activate gecko
           python applications/train_gecko_emulators.py -c config/ci_config.yml
-      - name: Run the gecko box model script
+      - name: Run the gecko box MLP model script
         run: |
           conda activate gecko
           python applications/run_gecko_emulators.py -c config/ci_config.yml
+        - name: Run the gecko train script for the GRU model
```
Comment (charlie-becker) [inline .github/workflows/train_and_run.yml:39]: Looks like there is improper YAML syntax here that prevented the CI from running.

**Review by charlie-becker — CHANGES_REQUESTED** [review]
Thanks for adding this. Please correct YAML syntax in the Github Actions workflow YAML so the CI will properly run.

File: .github/workflows/train_and_run.yml (hunk 1 above)
Comment (jsschreck) [reply .github/workflows/train_and_run.yml:39]: The sad/funny thing is I watched this fail but was already out the door and onto break 😂

will fix then merge!

**Comment by djgagne** [comment]
Try removing `"geckoml/torch"` from the setup call in setup.py.

**Review by djgagne — APPROVED** [review]
I don't see any glaring issues with the code, and it looks like the CI test passed after fixing setup.py. I only have a general style comment about `train_gecko_rnn_emulators.py`. Currently the optimizer and training loop are separate functions/classes dissociated from the model class, which can make it a little hard to follow the full logic of the class. The training loop is also in the application rather than in a built in fit routine. I personally prefer the R/scikit-learn/keras style of model classing where a fit method is included with the model class. Having a consistent set of methods/APIs across models makes it a lot easier to drop-in other models in the same data pipeline. Having 100% consistency is not always possible, but 90% consistency with a few clearly defined exceptions can save us all a lot of time in understanding each others' code.

**Review by charlie-becker — APPROVED** [review]

**Comment by charlie-becker** [comment]
Looks good
